app/controllers/user_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UserController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @users = User.where.not(id: current_user.id)
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    @user.admin_id = current_user.id if current_user.role == "Admin"
```

Comment:
ew`current_user.role == "Admin"` this condition checking was repeated at many places even in vi. I don't remember the PR but I would suggest to create a method `is_admin` in ApplicationController. @PragatiDoshi-BTC has created similar method in View helper. 

Co-ordinate with her and extract common method in Application helper and make this method to avail to view helper as well.